packages/block-editor/src/components/block-alignment-control/use-available-alignments.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { useLayout } from '../block-list/layout';
+import { store as blockEditorStore } from '../../store';
+
+const DEFAULT_CONTROLS = [ 'left', 'center', 'right', 'wide', 'full' ];
+const WIDE_CONTROLS = [ 'wide', 'full' ];
+
+export default function useAvailableAlignments( controls = DEFAULT_CONTROLS ) {
+	const { wideControlsEnabled = false } = useSelect( ( select ) => {
+		const { getSettings } = select( blockEditorStore );
+		const settings = getSettings();
+		return {
+			wideControlsEnabled: settings.alignWide,
+		};
+	}, [] );
+	const layout = useLayout();
+	const supportsAlignments = layout.type === 'default';
+
+	if ( ! supportsAlignments ) {
+		return [];
+	}
+	const { alignments: availableAlignments = DEFAULT_CONTROLS } = layout;
+	const enabledControls = controls.filter(
+		( control ) =>
+			( layout.alignments || // Ignore the global wideAlignment check if the layout explicitely defines alignments.
```

Comment:
why not use `alignments` from line 29?

Reply:
this is just the same old code moved to a separate file, I didn't think about its content too much.

Reply:
Oh I think the reason here is that we want to ignore the default value.